diff --git a/src/tm_mad/nfs/tm_context.sh b/src/tm_mad/nfs/tm_context.sh
index b6c515d..070a799 100755
--- a/src/tm_mad/nfs/tm_context.sh
+++ b/src/tm_mad/nfs/tm_context.sh
@@ -48,6 +48,12 @@ ISO_DIR=$DST_DIR/isofiles
 
 exec_and_log "mkdir -p $ISO_DIR"
 
+# Add StratusLab init script
+SRC="$SRC /var/share/stratuslab/context/init.sh"
+# Hack, can't get $VMDIR to work
+_VMDIR=`dirname $DST_DIR`
+SRC="$SRC $_VMDIR/internal_key.pub"
+
 for f in $SRC; do
     case $f in
     http://*)
diff --git a/src/tm_mad/ssh/tm_context.sh b/src/tm_mad/ssh/tm_context.sh
index b420be3..52f5194 100755
--- a/src/tm_mad/ssh/tm_context.sh
+++ b/src/tm_mad/ssh/tm_context.sh
@@ -48,6 +48,12 @@ ISO_DIR="$TMP_DIR/isofiles"
 
 exec_and_log "mkdir -p $ISO_DIR"
 
+# Add StratusLab init script
+SRC="$SRC /var/share/stratuslab/context/init.sh"
+# Hack, can't get $VMDIR to work
+_VMDIR=`dirname $DST_DIR`
+SRC="$SRC $_VMDIR/internal_key.pub"
+
 for f in $SRC; do
     case $f in
     http://*)
