diff --recursive --unified --unidirectional-new-file checkout/.git/logs/HEAD checkout-new/.git/logs/HEAD
--- checkout/src/authm_mad/exterior_auth.rb	1970-01-01 01:00:00.000000000 +0100
+++ checkout-new/src/authm_mad/exterior_auth.rb	2010-12-19 09:12:24.000000000 +0100
@@ -0,0 +1,37 @@
+#
+# Created as part of the StratusLab project (http://stratuslab.eu),
+# co-funded by the European Commission under the Grant Agreement
+# INSFO-RI-261552.
+#
+# Copyright (c) 2010, Centre Nationale de la Recherche Scientifique
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# 
+# Module for implementing exterior authentication.
+#
+# This modules simply returns true to authenticate all users.
+#
+# CAUTION: Exterior authentication mechanisms MUST be in place,
+# for example, via the authentication proxy.  
+#
+class ExteriorAuth
+    #
+    # Always responds true to the request!
+    # Ensure external authentication tools are in place.
+    #
+    def auth(user_id, user, password, token)
+        true
+    end
+end
