diff --git a/install.sh b/install.sh
index 58a0197..bad2023 100755
--- a/install.sh
+++ b/install.sh
@@ -311,6 +311,7 @@ RUBY_LIB_FILES="src/mad/ruby/ActionManager.rb \
                 src/authm_mad/one_usage.rb \
                 src/authm_mad/quota.rb \
                 src/authm_mad/simple_auth.rb \
+                src/authm_mad/exterior_auth.rb \
                 src/authm_mad/simple_permissions.rb \
                 src/authm_mad/ssh_auth.rb"
 
