diff --git a/src/tm_mad/ssh/tm_context.sh b/src/tm_mad/ssh/tm_context.sh
index b420be3..aab92f9 100755
--- a/src/tm_mad/ssh/tm_context.sh
+++ b/src/tm_mad/ssh/tm_context.sh
@@ -42,7 +42,13 @@ DST_PATH=`arg_path $DST`
 DST_DIR=`dirname $DST_PATH`
 DST_FILE=`basename $DST_PATH`
 DST_HASH=`echo -n $DST | $MD5SUM | $AWK '{print $1}'`
-TMP_DIR="$ONE_LOCATION/var/$DST_HASH"
+
+if [ -z "${ONE_LOCATION}" ]; then
+    TMP_DIR="/var/lib/one/var/$DST_HASH"
+else
+    TMP_DIR="$ONE_LOCATION/var/$DST_HASH"
+fi
+
 ISO_DIR="$TMP_DIR/isofiles"
 
