diff --git a/include/RequestManager.h b/include/RequestManager.h
index 9335054..0e734b2 100644
--- a/include/RequestManager.h
+++ b/include/RequestManager.h
@@ -256,7 +256,7 @@ private:
      *    @param object over which the action was applied
      *    @param id id of the object, -1 for Pool, -2 for no-id objects
      *              (allocate error, parse error)
-     *    @param rc returned error code (NULL to ignore)
+     *    @param rc returned error code (0 to ignore)
      *    @returns string for logging
      */
     static string action_error (const string& method,
