diff --git a/src/vmm_mad/exec/vmm_exec_kvm.conf b/src/vmm_mad/exec/vmm_exec_kvm.conf
index 949eae1..c25c1a7 100644
--- a/src/vmm_mad/exec/vmm_exec_kvm.conf
+++ b/src/vmm_mad/exec/vmm_exec_kvm.conf
@@ -31,7 +31,7 @@
 #VCPU = 1
 #MEMORY =  128
 
-OS       = [ boot = "hd", arch = "i686" ]
+OS       = [ boot = "hd", arch = "x86_64" ]
 FEATURES = [ PAE = "no", ACPI = "yes" ]
 
 DISK     = [ driver = "raw" , cache = "default"]
