diff --git a/src/tm_mad/ssh/tm_context.sh b/src/tm_mad/ssh/tm_context.sh
index 772b549..38c9292 100755
--- a/src/tm_mad/ssh/tm_context.sh
+++ b/src/tm_mad/ssh/tm_context.sh
@@ -52,6 +52,11 @@ ISO_DIR="$TMP_DIR/isofiles"
 exec_and_log "mkdir -p $ISO_DIR" \
     "Error creating directory $ISO_DIR"
 
+# Add StratusLab init scripts
+SRC="/var/share/stratuslab/context/init.sh $SRC"
+SRC="/var/share/stratuslab/context/epilog.sh $SRC"
+SRC="/var/share/stratuslab/context/prolog.sh $SRC"
+
 for f in $SRC; do
     case $f in
     http://*)
