diff --git a/install.sh b/install.sh
index 4dcf332..086c967 100755
--- a/install.sh
+++ b/install.sh
@@ -245,7 +245,8 @@ VAR_DIRS="$VAR_LOCATION/remotes \
           $VAR_LOCATION/remotes/auth/server_x509 \
           $VAR_LOCATION/remotes/auth/server_cipher \
           $VAR_LOCATION/remotes/auth/quota \
-          $VAR_LOCATION/remotes/auth/dummy"
+          $VAR_LOCATION/remotes/auth/dummy \
+          $VAR_LOCATION/remotes/auth/default"
 
 SUNSTONE_DIRS="$SUNSTONE_LOCATION/models \
                $SUNSTONE_LOCATION/models/OpenNebulaJSON \
@@ -375,6 +376,7 @@ INSTALL_FILES=(
     AUTH_SERVER_X509_FILES:$VAR_LOCATION/remotes/auth/server_x509
     AUTH_SERVER_CIPHER_FILES:$VAR_LOCATION/remotes/auth/server_cipher
     AUTH_DUMMY_FILES:$VAR_LOCATION/remotes/auth/dummy
+    AUTH_DEFAULT_FILES:$VAR_LOCATION/remotes/auth/default
     AUTH_PLAIN_FILES:$VAR_LOCATION/remotes/auth/plain
     AUTH_QUOTA_FILES:$VAR_LOCATION/remotes/auth/quota
     VMM_EXEC_KVM_SCRIPTS:$VAR_LOCATION/remotes/vmm/kvm
@@ -712,6 +714,8 @@ AUTH_SSH_FILES="src/authm_mad/remotes/ssh/authenticate"
 
 AUTH_DUMMY_FILES="src/authm_mad/remotes/dummy/authenticate"
 
+AUTH_DEFAULT_FILES="src/authm_mad/remotes/default/authenticate"
+
 AUTH_PLAIN_FILES="src/authm_mad/remotes/plain/authenticate"
 
 AUTH_QUOTA_FILES="src/authm_mad/remotes/quota/authorize"
@@ -1504,6 +1508,9 @@ if [ "$UNINSTALL" = "no" ] ; then
     if [ -d "$DESTDIR$IMAGES_LOCATION" ]; then
         chmod 3770 $DESTDIR$IMAGES_LOCATION
     fi
+
+    # Default auth script
+    chmod a+x $DESTDIR$VAR_LOCATION/remotes/auth/default/authenticate
 else
     for d in `echo $DELETE_DIRS | awk '{for (i=NF;i>=1;i--) printf $i" "}'`; do
         rmdir $d
diff --git a/src/authm_mad/remotes/default/authenticate b/src/authm_mad/remotes/default/authenticate
new file mode 100755
index 0000000..66c50a8
--- /dev/null
+++ b/src/authm_mad/remotes/default/authenticate
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+# -------------------------------------------------------------------------- #
+# Copyright 2002-2012, OpenNebula Project Leads (OpenNebula.org)             #
+#                                                                            #
+# Licensed under the Apache License, Version 2.0 (the "License"); you may    #
+# not use this file except in compliance with the License. You may obtain    #
+# a copy of the License at                                                   #
+#                                                                            #
+# http://www.apache.org/licenses/LICENSE-2.0                                 #
+#                                                                            #
+# Unless required by applicable law or agreed to in writing, software        #
+# distributed under the License is distributed on an "AS IS" BASIS,          #
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   #
+# See the License for the specific language governing permissions and        #
+# limitations under the License.                                             #
+#--------------------------------------------------------------------------- #
+
+# $1 = username
+# $2 = "-" if user is not registered in opennebula
+# $3 = password
+echo dummy $1 $3
+
+
+
+
diff --git a/src/authm_mad/remotes/dummy/authenticate b/src/authm_mad/remotes/dummy/authenticate
index 0917ac6..66c50a8 100755
--- a/src/authm_mad/remotes/dummy/authenticate
+++ b/src/authm_mad/remotes/dummy/authenticate
@@ -19,7 +19,7 @@
 # $1 = username
 # $2 = "-" if user is not registered in opennebula
 # $3 = password
-echo core $1 $3
+echo dummy $1 $3
 
 
 
