commit 07a0320a66226ba28d5570acc50c533994610d5d
Author: Javi Fontan <jfontan@gmail.com>
Date:   Wed Apr 11 12:59:05 2012 +0200

    stratuslab: enable source disk parameter for users

diff --git a/src/vm/VirtualMachineTemplate.cc b/src/vm/VirtualMachineTemplate.cc
index cfac858..58e1fa4 100644
--- a/src/vm/VirtualMachineTemplate.cc
+++ b/src/vm/VirtualMachineTemplate.cc
@@ -22,13 +22,13 @@
 
 const string VirtualMachineTemplate::RESTRICTED_ATTRIBUTES[] = {
 	"CONTEXT/FILES",
-	"DISK/SOURCE",
+//	"DISK/SOURCE",
 	"NIC/MAC",
 	"NIC/VLAN_ID",
 	"RANK"
 };
 
-const int VirtualMachineTemplate::RS_ATTRS_LENGTH = 5;
+const int VirtualMachineTemplate::RS_ATTRS_LENGTH = 4;
 	
 /* -------------------------------------------------------------------------- */
 /* -------------------------------------------------------------------------- */
