diff --git a/src/vmm/LibVirtDriverKVM.cc b/src/vmm/LibVirtDriverKVM.cc
index f40ebbc..70334fb 100644
--- a/src/vmm/LibVirtDriverKVM.cc
+++ b/src/vmm/LibVirtDriverKVM.cc
@@ -242,7 +242,7 @@ int LibVirtDriver::deployment_description_kvm(
 
     if (emulator == "kvm")
     {
-        file << "\t\t<emulator>/usr/bin/kvm</emulator>" << endl;
+        file << "\t\t<emulator>/usr/libexec/qemu-kvm</emulator>" << endl;
     }
 
     num = vm->get_template_attribute("DISK",attrs);
