diff --git a/src/vmm_mad/remotes/kvm/kvmrc b/src/vmm_mad/remotes/kvm/kvmrc
index ef89387..8ce0f4a 100644
--- a/src/vmm_mad/remotes/kvm/kvmrc
+++ b/src/vmm_mad/remotes/kvm/kvmrc
@@ -18,7 +18,7 @@ export LANG=C
 
 export LIBVIRT_URI=qemu:///system
 
-export QEMU_PROTOCOL=qemu
+export QEMU_PROTOCOL=qemu+ssh
 
 #------------------------------------------------------------------------------
 # Wait the VM to shutdown TIMEOUT (POLL_INTERVAL) seconds.
