diff --git a/src/tm_mad/tm_common.sh b/src/tm_mad/tm_common.sh
index cb087cc..a7dac33 100644
--- a/src/tm_mad/tm_common.sh
+++ b/src/tm_mad/tm_common.sh
@@ -55,7 +55,8 @@ function get_vmdir
 # /some//path///somewhere/ -> /some/path/somewhere
 function fix_dir_slashes
 {
-    dirname "$1/file" | $SED 's/\/+/\//g'
+    echo ""
+#    dirname "$1/file" | $SED 's/\/+/\//g'
 }
 
 function fix_var_slashes
