diff --git a/src/tm_mad/nfs/tm_nfs.conf b/src/tm_mad/nfs/tm_nfs.conf
index 5158752..ec1397e 100644
--- a/src/tm_mad/nfs/tm_nfs.conf
+++ b/src/tm_mad/nfs/tm_nfs.conf
@@ -2,6 +2,6 @@ CLONE   = nfs/tm_clone.sh
 LN      = nfs/tm_ln.sh
 MKSWAP  = nfs/tm_mkswap.sh
 MKIMAGE = nfs/tm_mkimage.sh
-DELETE  = nfs/tm_delete.sh
+DELETE  = nfs/tm_quarantine.sh
 MV      = nfs/tm_mv.sh
 CONTEXT = nfs/tm_context.sh
