diff --git a/src/template/Template.cc b/src/template/Template.cc
index b1dd8b5..f483f2e 100644
--- a/src/template/Template.cc
+++ b/src/template/Template.cc
@@ -14,8 +14,6 @@
 /* limitations under the License.                                             */
 /* -------------------------------------------------------------------------- */
 
-#include <sqlite3.h>
-
 #include "Template.h"
 #include "template_syntax.h"
 
