diff --git a/share/examples/tm/tm_clone.sh b/share/examples/tm/tm_clone.sh
index 098c246..b4a7ffe 100755
--- a/share/examples/tm/tm_clone.sh
+++ b/share/examples/tm/tm_clone.sh
@@ -38,7 +38,7 @@ DST_DIR=`dirname $DST_PATH`
 
 log "Creating directory $DST_DIR"
 exec_and_log "ssh $DST_HOST mkdir -p $DST_DIR"
-exec_and_log "ssh $DST_HOST chmod a+w $DST_DIR"
+exec_and_log "ssh $DST_HOST chmod ug+w,o-rwx $DST_DIR"
 
 case $SRC in
 http://*)
@@ -52,5 +52,5 @@ http://*)
     ;;
 esac
 
-exec_and_log "ssh $DST_HOST chmod a+w $DST_PATH"
+exec_and_log "ssh $DST_HOST chmod ug+w,o-rwx $DST_PATH"
 
diff --git a/share/examples/tm/tm_mkswap.sh b/share/examples/tm/tm_mkswap.sh
index ea98602..3f68e83 100755
--- a/share/examples/tm/tm_mkswap.sh
+++ b/share/examples/tm/tm_mkswap.sh
@@ -36,5 +36,5 @@ exec_and_log "ssh $DST_HOST dd if=/dev/zero of=$DST_PATH bs=1 count=1 seek=${SIZ
 log "Initializing swap space"
 exec_and_log "ssh $DST_HOST mkswap $DST_PATH"
 
-exec_and_log "ssh $DST_HOST chmod a+w $DST_PATH"
+exec_and_log "ssh $DST_HOST chmod ug+w,o-rwx $DST_PATH"
 
diff --git a/src/mad/sh/scripts_common.sh b/src/mad/sh/scripts_common.sh
index 3d59047..6ac998e 100755
--- a/src/mad/sh/scripts_common.sh
+++ b/src/mad/sh/scripts_common.sh
@@ -32,6 +32,9 @@ SED=/bin/sed
 SSH=/usr/bin/ssh
 SUDO=/usr/bin/sudo
 WGET=/usr/bin/wget
+CURL=/usr/bin/curl
+GUNZIP=/bin/gunzip
+BUNZIP2=/bin/bunzip2
 
 # Used for log messages
 SCRIPT_NAME=`basename $0`
@@ -145,3 +148,19 @@ function timeout_exec_and_log
         exit $CMD_CODE
     fi
 }
+
+function download_and_extract
+{
+    URL=$1
+    ARCHIVE=$2
+    EXTRACT_CMD=$3
+    HOST=$4
+
+    if [ "x$HOST" != "x" ]; then
+        SSH_CMD="$SSH $HOST"
+    fi
+
+    $SSH_CMD $CURL -k -o $ARCHIVE $URL
+    $SSH_CMD $EXTRACT_CMD $ARCHIVE
+}
+
diff --git a/src/tm_mad/lvm/tm_context.sh b/src/tm_mad/lvm/tm_context.sh
index ee7fd40..887d978 100755
--- a/src/tm_mad/lvm/tm_context.sh
+++ b/src/tm_mad/lvm/tm_context.sh
@@ -62,6 +62,7 @@ for f in $SRC; do
 done
 
 exec_and_log "$MKISOFS -o $TMP_DIR/$DST_FILE -J -R $ISO_DIR"
+exec_and_log "chmod o-rwx $TMP_DIR/$DST_FILE"
 exec_and_log "$SCP $TMP_DIR/$DST_FILE $DST"
 exec_and_log "rm -rf $TMP_DIR"
 
diff --git a/src/tm_mad/lvm/tm_mkimage.sh b/src/tm_mad/lvm/tm_mkimage.sh
index d092a8b..131cc17 100755
--- a/src/tm_mad/lvm/tm_mkimage.sh
+++ b/src/tm_mad/lvm/tm_mkimage.sh
@@ -35,4 +35,4 @@ DST_DIR=`dirname $DST_PATH`
 exec_and_log "$SSH $DST_HOST mkdir -p $DST_DIR"
 exec_and_log "$SSH $DST_HOST $DD if=/dev/zero of=$DST_PATH bs=1 count=1 seek=${SIZE}M"
 exec_and_log "$SSH $DST_HOST $MKFS -t $FSTYPE -F $DST_PATH"
-exec_and_log "$SSH $DST_HOST chmod a+rw $DST_PATH"
+exec_and_log "$SSH $DST_HOST chmod ug+rw,o-rwx $DST_PATH"
diff --git a/src/tm_mad/lvm/tm_mkswap.sh b/src/tm_mad/lvm/tm_mkswap.sh
index 046a277..17a0a6e 100755
--- a/src/tm_mad/lvm/tm_mkswap.sh
+++ b/src/tm_mad/lvm/tm_mkswap.sh
@@ -39,5 +39,5 @@ exec_and_log "$SSH $DST_HOST $DD if=/dev/zero of=$DST_PATH bs=1 count=1 seek=${S
 log "Initializing swap space"
 exec_and_log "$SSH $DST_HOST $MKSWAP $DST_PATH"
 
-exec_and_log "$SSH $DST_HOST chmod a+w $DST_PATH"
+exec_and_log "$SSH $DST_HOST chmod ug+w,o-rwx $DST_PATH"
 
diff --git a/src/tm_mad/nfs/tm_clone.sh b/src/tm_mad/nfs/tm_clone.sh
index 62c33fc..a80a83c 100755
--- a/src/tm_mad/nfs/tm_clone.sh
+++ b/src/tm_mad/nfs/tm_clone.sh
@@ -32,30 +32,64 @@ get_vmdir
 SRC_PATH=`arg_path $SRC`
 DST_PATH=`arg_path $DST`
 
+DST_HOST=`arg_host $DST`
+DST_DIR=`dirname $DST_PATH`
+
 fix_paths
 
 log_debug "$1 $2"
 log_debug "DST: $DST_PATH"
 
-DST_DIR=`dirname $DST_PATH`
 
 log "Creating directory $DST_DIR"
 exec_and_log "mkdir -p $DST_DIR"
-exec_and_log "chmod a+w $DST_DIR"
-
-case $SRC in
-http://*)
-    log "Downloading $SRC"
-    exec_and_log "$WGET -O $DST_PATH $SRC" \
-        "Error downloading $SRC"
-    ;;
-
-*)
-    log "Cloning $SRC_PATH"
-    exec_and_log "cp -r $SRC_PATH $DST_PATH" \
-        "Error copying $SRC to $DST"
-    ;;
-esac
-
-exec_and_log "chmod a+rw $DST_PATH"
+exec_and_log "chmod ug+w,o-rwx $DST_DIR"
+
+function _download_and_extract () {
+    case $SRC in
+    http://*.gz|https://*.gz|ftp://*.gz)
+        log "Downloading GZip archive $SRC"
+        exec_and_log "download_and_extract $SRC $DST_PATH.gz $GUNZIP" \
+            "Error downloading $SRC"
+        ;;
+
+    http://*.bz2|https://*.bz2|ftp://*.bz2)
+        log "Downloading BZip2 archive $SRC"
+        exec_and_log "download_and_extract $SRC $DST_PATH.bz2 $BUNZIP2" \
+            "Error downloading $SRC"
+        ;;
+
+    http://*|https://*|ftp://*)
+        LOCATION=$(stratus-manifest --get-element location $SRC)
+        rc=$?
+        if [ "$rc" == "0" ] && [ -n "$LOCATION" ]; then
+            log "Image location from image manifest: $LOCATION"
+            SRC=$LOCATION
+            _download_and_extract
+        else
+            log "Downloading $SRC"
+            exec_and_log "$CURL -k -o $DST_PATH $SRC" \
+                "Error downloading $SRC"
+        fi
+        ;;
+
+    pdisk:*)
+        log "Persistent disk handling $SRC $DST"
+        exec_and_log "$SSH -t -t $DST_HOST /usr/sbin/attach-persistent-disk.sh $SRC $DST_PATH" \
+            "Failed to clone persistent disk $DST_PATH"
+        ;;
+
+    *)
+        log "Cloning $SRC_PATH"
+        exec_and_log "cp -r $SRC_PATH $DST_PATH" \
+            "Error copying $SRC to $DST"
+        ;;
+    esac
+}
+
+_download_and_extract
+
+if [ -e $DST_PATH ] && [ ! -L $DST_PATH ]; then
+  exec_and_log "chmod ug+rw,o-rwx $DST_PATH"
+fi

diff --git a/src/tm_mad/nfs/tm_context.sh b/src/tm_mad/nfs/tm_context.sh
index f211024..ae44a47 100755
--- a/src/tm_mad/nfs/tm_context.sh
+++ b/src/tm_mad/nfs/tm_context.sh
@@ -62,6 +62,8 @@ done
 exec_and_log "$MKISOFS -o $DST_PATH -J -R $ISO_DIR" \
     "Error creating iso fs"
 
+exec_and_log "chmod o-rwx $DST_PATH"
+
 exec_and_log "rm -rf $ISO_DIR"
 
 
diff --git a/src/tm_mad/nfs/tm_ln.sh b/src/tm_mad/nfs/tm_ln.sh
index 1a3578e..e0254dc 100755
--- a/src/tm_mad/nfs/tm_ln.sh
+++ b/src/tm_mad/nfs/tm_ln.sh
@@ -39,7 +39,7 @@ DST_DIR=`dirname $DST_PATH`
 log "Creating directory $DST_DIR"
 exec_and_log "mkdir -p $DST_DIR" \
     "Could not create directory $DST_DIR"
-exec_and_log "chmod a+w $DST_DIR"
+exec_and_log "chmod ug+w,o-rwx $DST_DIR"
 
 log "Link $SRC_PATH"
 exec_and_log "ln -s $SRC_PATH $DST_PATH"
diff --git a/src/tm_mad/nfs/tm_mkimage.sh b/src/tm_mad/nfs/tm_mkimage.sh
index b30c0ff..f83be45 100755
--- a/src/tm_mad/nfs/tm_mkimage.sh
+++ b/src/tm_mad/nfs/tm_mkimage.sh
@@ -42,5 +42,5 @@ exec_and_log "$DD if=/dev/zero of=$DST_PATH bs=1 count=1 seek=${SIZE}M" \
     "Could not create image $DST_PATH"
 exec_and_log "$MKFS -t $FSTYPE -F $DST_PATH" \
     "Unable to create filesystem $FSTYPE in $DST_PATH"
-exec_and_log "chmod a+rw $DST_PATH"
+exec_and_log "chmod ug+rw,o-rwx $DST_PATH"
 
diff --git a/src/tm_mad/nfs/tm_mkswap.sh b/src/tm_mad/nfs/tm_mkswap.sh
index 58b1e28..c44e3a8 100755
--- a/src/tm_mad/nfs/tm_mkswap.sh
+++ b/src/tm_mad/nfs/tm_mkswap.sh
@@ -37,7 +37,7 @@ DST_DIR=`dirname $DST_PATH`
 
 log_debug "Creating directory $DST_DIR"
 exec_and_log "mkdir -p $DST_DIR"
-exec_and_log "chmod a+w $DST_DIR"
+exec_and_log "chmod ug+w,o-rwx $DST_DIR"
 
 log_debug "Creating ${SIZE}Mb image in $DST_PATH"
 exec_and_log "$DD if=/dev/zero of=$DST_PATH bs=1 count=1 seek=${SIZE}M" \
@@ -47,5 +47,5 @@ log_debug "Initializing swap space"
 exec_and_log "$MKSWAP $DST_PATH" \
     "Could not create swap on $DST_PATH"
 
-exec_and_log "chmod a+w $DST_PATH"
+exec_and_log "chmod ug+w,o-rwx $DST_PATH"
 
diff --git a/src/tm_mad/ssh/tm_clone.sh b/src/tm_mad/ssh/tm_clone.sh
index 71800d1..7cb067e 100755
--- a/src/tm_mad/ssh/tm_clone.sh
+++ b/src/tm_mad/ssh/tm_clone.sh
@@ -33,7 +33,6 @@ DST_PATH=`arg_path $DST`
 SRC_HOST=`arg_host $SRC`
 DST_HOST=`arg_host $DST`
 
-
 log_debug "$1 $2"
 log_debug "DST: $DST_PATH"
 
@@ -43,19 +42,50 @@ log "Creating directory $DST_DIR"
 exec_and_log "$SSH $DST_HOST mkdir -p $DST_DIR" \
     "Error creating directory $DST_DIR"
 
-case $SRC in
-http://*)
-    log "Downloading $SRC"
-    exec_and_log "$SSH $DST_HOST $WGET -O $DST_PATH $SRC" \
-        "Error downloading $SRC"
-    ;;
-
-*)
-    log "Cloning $SRC"
-    exec_and_log "$SCP $SRC $DST" \
-        "Error copying $SRC to $DST"
-    ;;
-esac
-
-exec_and_log "$SSH $DST_HOST chmod a+rw $DST_PATH"
+function _download_and_extract () {
+    case $SRC in
+    http://*.gz|https://*.gz|ftp://*.gz)
+        log "Downloading GZip archive $SRC"
+        exec_and_log "download_and_extract $SRC $DST_PATH.gz $GUNZIP $DST_HOST" \
+            "Error downloading $SRC"
+        ;;
+
+    http://*.bz2|https://*.bz2|ftp://*.bz2)
+        log "Downloading BZip2 archive $SRC"
+        exec_and_log "download_and_extract $SRC $DST_PATH.bz2 $BUNZIP2 $DST_HOST" \
+            "Error downloading $SRC"
+        ;;
+
+    http://*|https://*|ftp://*)
+        LOCATION=$(stratus-manifest --get-element location $SRC)
+        rc=$?
+        if [ "$rc" == "0" ] && [ -n "$LOCATION" ]; then
+            log "Image location from image manifest: $LOCATION"
+            SRC=$LOCATION
+            _download_and_extract
+        else
+            log "Downloading $SRC"
+            exec_and_log "$SSH $DST_HOST $CURL -k -o $DST_PATH $SRC" \
+                "Error downloading $SRC"
+        fi
+        ;;
+
+    pdisk:*)
+        log "Persistent disk handling $SRC $DST"
+        exec_and_log "$SSH -t -t $DST_HOST /usr/sbin/attach-persistent-disk.sh $SRC $DST_PATH" \
+            "Failed to clone persistent disk $DST_PATH"
+        ;;
+
+    *)
+        log "Cloning $SRC"
+        exec_and_log "$SCP $SRC $DST" "Error copying $SRC to $DST"
+        ;;
+    esac
+}
+
+_download_and_extract
+
+if [ ! -L $DST_PATH ]; then
+  exec_and_log "$SSH $DST_HOST chmod --quiet ug+w,o-rwx $DST_PATH"
+fi
 
diff --git a/src/tm_mad/ssh/tm_context.sh b/src/tm_mad/ssh/tm_context.sh
index 322ebda..bdff870 100755
--- a/src/tm_mad/ssh/tm_context.sh
+++ b/src/tm_mad/ssh/tm_context.sh
@@ -66,6 +66,7 @@ done
 
 exec_and_log "$MKISOFS -o $TMP_DIR/$DST_FILE -J -R $ISO_DIR" \
     "Error creating iso fs"
+exec_and_log "chmod o-rwx $TMP_DIR/$DST_FILE"
 exec_and_log "$SCP $TMP_DIR/$DST_FILE $DST" \
     "Error copying $TMP_DIR/$DST_FILE to $DST"
 exec_and_log "rm -rf $TMP_DIR" \
diff --git a/src/tm_mad/ssh/tm_mkimage.sh b/src/tm_mad/ssh/tm_mkimage.sh
index 81ff20f..63aefe3 100755
--- a/src/tm_mad/ssh/tm_mkimage.sh
+++ b/src/tm_mad/ssh/tm_mkimage.sh
@@ -38,4 +38,5 @@ exec_and_log "$SSH $DST_HOST $DD if=/dev/zero of=$DST_PATH bs=1 count=1 seek=${S
     "Could not create image $DST_PATH"
 exec_and_log "$SSH $DST_HOST $MKFS -t $FSTYPE -F $DST_PATH" \
     "Unable to create filesystem $FSTYPE in $DST_PATH"
-exec_and_log "$SSH $DST_HOST chmod a+rw $DST_PATH"
\ No newline at end of file
+exec_and_log "$SSH $DST_HOST chmod ug+rw,o-rwx $DST_PATH"
+
diff --git a/src/tm_mad/ssh/tm_mkswap.sh b/src/tm_mad/ssh/tm_mkswap.sh
index 6070513..5379eb9 100755
--- a/src/tm_mad/ssh/tm_mkswap.sh
+++ b/src/tm_mad/ssh/tm_mkswap.sh
@@ -41,5 +41,5 @@ log "Initializing swap space"
 exec_and_log "$SSH $DST_HOST $MKSWAP $DST_PATH" \
     "Could not create swap on $DST_PATH"
 
-exec_and_log "$SSH $DST_HOST chmod a+w $DST_PATH"
+exec_and_log "$SSH $DST_HOST chmod ug+w,o-rwx $DST_PATH"
 
