diff --git a/share/examples/tm/tm_clone.sh b/share/examples/tm/tm_clone.sh
index 098c246..b4a7ffe 100755
--- a/share/examples/tm/tm_clone.sh
+++ b/share/examples/tm/tm_clone.sh
@@ -38,7 +38,7 @@ DST_DIR=`dirname $DST_PATH`
 
 log "Creating directory $DST_DIR"
 exec_and_log "ssh $DST_HOST mkdir -p $DST_DIR"
-exec_and_log "ssh $DST_HOST chmod a+w $DST_DIR"
+exec_and_log "ssh $DST_HOST chmod ug+w,o-rwx $DST_DIR"
 
 case $SRC in
 http://*)
@@ -52,5 +52,5 @@ http://*)
     ;;
 esac
 
-exec_and_log "ssh $DST_HOST chmod a+w $DST_PATH"
+exec_and_log "ssh $DST_HOST chmod ug+w,o-rwx $DST_PATH"
 
diff --git a/share/examples/tm/tm_mkswap.sh b/share/examples/tm/tm_mkswap.sh
index ea98602..3f68e83 100755
--- a/share/examples/tm/tm_mkswap.sh
+++ b/share/examples/tm/tm_mkswap.sh
@@ -36,5 +36,5 @@ exec_and_log "ssh $DST_HOST dd if=/dev/zero of=$DST_PATH bs=1 count=1 seek=${SIZ
 log "Initializing swap space"
 exec_and_log "ssh $DST_HOST mkswap $DST_PATH"
 
-exec_and_log "ssh $DST_HOST chmod a+w $DST_PATH"
+exec_and_log "ssh $DST_HOST chmod ug+w,o-rwx $DST_PATH"
 
diff --git a/src/mad/sh/scripts_common.sh b/src/mad/sh/scripts_common.sh
index 3127108..9ab2bf4 100755
--- a/src/mad/sh/scripts_common.sh
+++ b/src/mad/sh/scripts_common.sh
@@ -32,6 +32,9 @@ SED=/bin/sed
 SSH=/usr/bin/ssh
 SUDO=/usr/bin/sudo
 WGET=/usr/bin/wget
+CURL=/usr/bin/curl
+GUNZIP=/bin/gunzip
+BUNZIP2=/bin/bunzip2
 
 # Used for log messages
 SCRIPT_NAME=`basename $0`
@@ -117,3 +120,19 @@ function timeout_exec_and_log
         exit $CMD_CODE
     fi
 }
+
+function download_and_extract
+{
+    URL=$1
+    ARCHIVE=$2
+    EXTRACT_CMD=$3
+    HOST=$4
+
+    if [ "x$HOST" != "x" ]; then
+        SSH_CMD="$SSH $HOST"
+    fi
+
+    $SSH_CMD $CURL -k -o $ARCHIVE $URL
+    $SSH_CMD $EXTRACT_CMD $ARCHIVE
+}
+
diff --git a/src/tm_mad/lvm/tm_context.sh b/src/tm_mad/lvm/tm_context.sh
index ee7fd40..887d978 100755
--- a/src/tm_mad/lvm/tm_context.sh
+++ b/src/tm_mad/lvm/tm_context.sh
@@ -62,6 +62,7 @@ for f in $SRC; do
 done
 
 exec_and_log "$MKISOFS -o $TMP_DIR/$DST_FILE -J -R $ISO_DIR"
+exec_and_log "chmod o-rwx $TMP_DIR/$DST_FILE"
 exec_and_log "$SCP $TMP_DIR/$DST_FILE $DST"
 exec_and_log "rm -rf $TMP_DIR"
 
diff --git a/src/tm_mad/lvm/tm_mkimage.sh b/src/tm_mad/lvm/tm_mkimage.sh
index d092a8b..131cc17 100755
--- a/src/tm_mad/lvm/tm_mkimage.sh
+++ b/src/tm_mad/lvm/tm_mkimage.sh
@@ -35,4 +35,4 @@ DST_DIR=`dirname $DST_PATH`
 exec_and_log "$SSH $DST_HOST mkdir -p $DST_DIR"
 exec_and_log "$SSH $DST_HOST $DD if=/dev/zero of=$DST_PATH bs=1 count=1 seek=${SIZE}M"
 exec_and_log "$SSH $DST_HOST $MKFS -t $FSTYPE -F $DST_PATH"
-exec_and_log "$SSH $DST_HOST chmod a+rw $DST_PATH"
+exec_and_log "$SSH $DST_HOST chmod ug+rw,o-rwx $DST_PATH"
diff --git a/src/tm_mad/lvm/tm_mkswap.sh b/src/tm_mad/lvm/tm_mkswap.sh
index 046a277..17a0a6e 100755
--- a/src/tm_mad/lvm/tm_mkswap.sh
+++ b/src/tm_mad/lvm/tm_mkswap.sh
@@ -39,5 +39,5 @@ exec_and_log "$SSH $DST_HOST $DD if=/dev/zero of=$DST_PATH bs=1 count=1 seek=${S
 log "Initializing swap space"
 exec_and_log "$SSH $DST_HOST $MKSWAP $DST_PATH"
 
-exec_and_log "$SSH $DST_HOST chmod a+w $DST_PATH"
+exec_and_log "$SSH $DST_HOST chmod ug+w,o-rwx $DST_PATH"
 
diff --git a/src/tm_mad/nfs/tm_clone.sh b/src/tm_mad/nfs/tm_clone.sh
index b54c826..e08c9e4 100755
--- a/src/tm_mad/nfs/tm_clone.sh
+++ b/src/tm_mad/nfs/tm_clone.sh
@@ -39,21 +39,50 @@ log "DST: $DST_PATH"
 
 DST_DIR=`dirname $DST_PATH`
 
+
 log "Creating directory $DST_DIR"
 exec_and_log "mkdir -p $DST_DIR"
-exec_and_log "chmod a+w $DST_DIR"
-
-case $SRC in
-http://*)
-    log "Downloading $SRC"
-    exec_and_log "$WGET -O $DST_PATH $SRC"
-    ;;
+exec_and_log "chmod ug+w,o-rwx $DST_DIR"
 
-*)
-    log "Cloning $SRC_PATH"
-    exec_and_log "cp -r $SRC_PATH $DST_PATH"
-    ;;
-esac
+function _download_and_extract () {
+    case $SRC in
+    http://*.gz|https://*.gz)
+        log "Downloading GZip archive $SRC"
+        exec_and_log "download_and_extract $SRC $DST_PATH.gz $GUNZIP"
+        ;;
+    
+    http://*.bz2|https://*.bz2)
+        log "Downloading BZip2 archive $SRC"
+        exec_and_log "download_and_extract $SRC $DST_PATH.bz2 $BUNZIP2"
+        ;;
+    
+    http://*|https://)
+        LOCATION=$(python2.6 $(which stratus-manifest) --get-element location $SRC)
+        rc=$?
+        if [ "$rc" == "0" ] && [ -n "$LOCATION" ]; then
+            log "Image location from image manifest: $LOCATION"
+            SRC=$LOCATION
+            _download_and_extract
+        else
+            log "Downloading $SRC"
+            exec_and_log "$CURL -k -o $DST_PATH $SRC"
+        fi
+        ;;
+    
+    pdisk:*)
+        log "Persistent disk handling $SRC $DST"
+        exec_and_log "sudo /usr/sbin/attach-persistent-disk.sh $SRC $DST_PATH"
+        ;;
+    
+    *)
+        log "Cloning $SRC_PATH"
+        exec_and_log "cp -r $SRC_PATH $DST_PATH"
+        ;;
+    esac
+}
 
-exec_and_log "chmod a+rw $DST_PATH"
+_download_and_extract
 
+if [ -e $DST_PATH && ! -L $DST_PATH ]; then
+  exec_and_log "chmod ug+rw,o-rwx $DST_PATH"
+fi
diff --git a/src/tm_mad/nfs/tm_context.sh b/src/tm_mad/nfs/tm_context.sh
index b079622..ac6c2b7 100755
--- a/src/tm_mad/nfs/tm_context.sh
+++ b/src/tm_mad/nfs/tm_context.sh
@@ -59,6 +59,8 @@ done
 
 exec_and_log "$MKISOFS -o $DST_PATH -J -R $ISO_DIR"
 
+exec_and_log "chmod o-rwx $DST_PATH"
+
 exec_and_log "rm -rf $ISO_DIR"
 
 
diff --git a/src/tm_mad/nfs/tm_ln.sh b/src/tm_mad/nfs/tm_ln.sh
index 33b4a2a..8b6ab90 100755
--- a/src/tm_mad/nfs/tm_ln.sh
+++ b/src/tm_mad/nfs/tm_ln.sh
@@ -38,7 +38,7 @@ DST_DIR=`dirname $DST_PATH`
 
 log "Creating directory $DST_DIR"
 exec_and_log "mkdir -p $DST_DIR"
-exec_and_log "chmod a+w $DST_DIR"
+exec_and_log "chmod ug+w,o-rwx $DST_DIR"
 
 log "Link $SRC_PATH"
 exec_and_log "ln -s $SRC_PATH $DST_PATH"
diff --git a/src/tm_mad/nfs/tm_mkimage.sh b/src/tm_mad/nfs/tm_mkimage.sh
index ddafb40..3184025 100755
--- a/src/tm_mad/nfs/tm_mkimage.sh
+++ b/src/tm_mad/nfs/tm_mkimage.sh
@@ -39,5 +39,5 @@ DST_DIR=`dirname $DST_PATH`
 exec_and_log "mkdir -p $DST_DIR"
 exec_and_log "$DD if=/dev/zero of=$DST_PATH bs=1 count=1 seek=${SIZE}M"
 exec_and_log "$MKFS -t $FSTYPE -F $DST_PATH"
-exec_and_log "chmod a+rw $DST_PATH"
+exec_and_log "chmod ug+rw,o-rwx $DST_PATH"
 
diff --git a/src/tm_mad/nfs/tm_mkswap.sh b/src/tm_mad/nfs/tm_mkswap.sh
index af78a59..635a92a 100755
--- a/src/tm_mad/nfs/tm_mkswap.sh
+++ b/src/tm_mad/nfs/tm_mkswap.sh
@@ -37,7 +37,7 @@ DST_DIR=`dirname $DST_PATH`
 
 log "Creating directory $DST_DIR"
 exec_and_log "mkdir -p $DST_DIR"
-exec_and_log "chmod a+w $DST_DIR"
+exec_and_log "chmod ug+w,o-rwx $DST_DIR"
 
 log "Creating ${SIZE}Mb image in $DST_PATH"
 exec_and_log "$DD if=/dev/zero of=$DST_PATH bs=1 count=1 seek=${SIZE}M"
@@ -45,5 +45,5 @@ exec_and_log "$DD if=/dev/zero of=$DST_PATH bs=1 count=1 seek=${SIZE}M"
 log "Initializing swap space"
 exec_and_log "$MKSWAP $DST_PATH"
 
-exec_and_log "chmod a+w $DST_PATH"
+exec_and_log "chmod ug+w,o-rwx $DST_PATH"
 
diff --git a/src/tm_mad/ssh/tm_clone.sh b/src/tm_mad/ssh/tm_clone.sh
index 8507c43..216370c 100755
--- a/src/tm_mad/ssh/tm_clone.sh
+++ b/src/tm_mad/ssh/tm_clone.sh
@@ -42,17 +42,47 @@ DST_DIR=`dirname $DST_PATH`
 log "Creating directory $DST_DIR"
 exec_and_log "$SSH $DST_HOST mkdir -p $DST_DIR"
 
-case $SRC in
-http://*)
-    log "Downloading $SRC"
-    exec_and_log "$SSH $DST_HOST $WGET -O $DST_PATH $SRC"
-    ;;
+function _download_and_extract () {
+    case $SRC in
+    http://*.gz|https://*.gz)
+        log "Downloading GZip archive $SRC"
+        exec_and_log "download_and_extract $SRC $DST_PATH.gz $GUNZIP $DST_HOST"
+        ;;
+    
+    http://*.bz2|https://*.bz2)
+        log "Downloading BZip2 archive $SRC"
+        exec_and_log "download_and_extract $SRC $DST_PATH.bz2 $BUNZIP2 $DST_HOST"
+        ;;
+    
+    http://*|https://)
+        LOCATION=$(python2.6 $(which stratus-manifest) --get-element location $SRC)
+        rc=$?
+        if [ "$rc" == "0" ] && [ -n "$LOCATION" ]; then
+            log "Image location from image manifest: $LOCATION"
+            SRC=$LOCATION
+            _download_and_extract
+        else
+            log "Downloading $SRC"
+            exec_and_log "$SSH $DST_HOST $CURL -k -o $DST_PATH $SRC"
+        fi
+        ;; 
 
-*)
-    log "Cloning $SRC"
-    exec_and_log "$SCP $SRC $DST"
-    ;;
-esac
+    pdisk:*)
+        log "Persistent disk handling $SRC $DST"
+        DST_HOST=`arg_host $DST`
+        exec_and_log "$SSH -t -t $DST_HOST sudo /usr/sbin/attach-persistent-disk.sh $SRC $DST_PATH"
+        ;;
 
-exec_and_log "$SSH $DST_HOST chmod a+rw $DST_PATH"
+    
+    *)
+        log "Cloning $SRC"
+        exec_and_log "$SCP $SRC $DST"
+        ;;
+    esac
+}
 
+_download_and_extract
+
+if [ ! -L $DST_PATH ]; then
+  exec_and_log "$SSH $DST_HOST chmod --quiet ug+w,o-rwx $DST_PATH"
+fi
diff --git a/src/tm_mad/ssh/tm_context.sh b/src/tm_mad/ssh/tm_context.sh
index ee7fd40..887d978 100755
--- a/src/tm_mad/ssh/tm_context.sh
+++ b/src/tm_mad/ssh/tm_context.sh
@@ -62,6 +62,7 @@ for f in $SRC; do
 done
 
 exec_and_log "$MKISOFS -o $TMP_DIR/$DST_FILE -J -R $ISO_DIR"
+exec_and_log "chmod o-rwx $TMP_DIR/$DST_FILE"
 exec_and_log "$SCP $TMP_DIR/$DST_FILE $DST"
 exec_and_log "rm -rf $TMP_DIR"
 
diff --git a/src/tm_mad/ssh/tm_mkimage.sh b/src/tm_mad/ssh/tm_mkimage.sh
index d092a8b..131cc17 100755
--- a/src/tm_mad/ssh/tm_mkimage.sh
+++ b/src/tm_mad/ssh/tm_mkimage.sh
@@ -35,4 +35,4 @@ DST_DIR=`dirname $DST_PATH`
 exec_and_log "$SSH $DST_HOST mkdir -p $DST_DIR"
 exec_and_log "$SSH $DST_HOST $DD if=/dev/zero of=$DST_PATH bs=1 count=1 seek=${SIZE}M"
 exec_and_log "$SSH $DST_HOST $MKFS -t $FSTYPE -F $DST_PATH"
-exec_and_log "$SSH $DST_HOST chmod a+rw $DST_PATH"
+exec_and_log "$SSH $DST_HOST chmod ug+rw,o-rwx $DST_PATH"
diff --git a/src/tm_mad/ssh/tm_mkswap.sh b/src/tm_mad/ssh/tm_mkswap.sh
index 046a277..17a0a6e 100755
--- a/src/tm_mad/ssh/tm_mkswap.sh
+++ b/src/tm_mad/ssh/tm_mkswap.sh
@@ -39,5 +39,5 @@ exec_and_log "$SSH $DST_HOST $DD if=/dev/zero of=$DST_PATH bs=1 count=1 seek=${S
 log "Initializing swap space"
 exec_and_log "$SSH $DST_HOST $MKSWAP $DST_PATH"
 
-exec_and_log "$SSH $DST_HOST chmod a+w $DST_PATH"
+exec_and_log "$SSH $DST_HOST chmod ug+w,o-rwx $DST_PATH"
 
