diff --git a/src/mad/sh/scripts_common.sh b/src/mad/sh/scripts_common.sh
index 79d45dd..1b21e3c 100755
--- a/src/mad/sh/scripts_common.sh
+++ b/src/mad/sh/scripts_common.sh
@@ -16,25 +16,28 @@
 
 # Paths for utilities
 export PATH=/bin:/sbin:/usr/bin:$PATH
-AWK=awk
-BASH=bash
-CUT=cut
-DATE=date
-DD=dd
-DU=du
-LVCREATE=lvcreate
-LVREMOVE=lvremove
-LVS=lvs
-MD5SUM=md5sum
-MKFS=mkfs
-MKISOFS=genisoimage
-MKSWAP=mkswap
-SCP=scp
-SED=sed
-SSH=ssh
-SUDO=sudo
-WGET=wget
-READLINK=readlink
+AWK=/bin/awk
+BASH=/bin/bash
+CUT=/bin/cut
+DATE=/bin/date
+DD=/bin/dd
+DU=/usr/bin/du
+LVCREATE=/sbin/lvcreate
+LVREMOVE=/sbin/lvremove
+LVS=/sbin/lvs
+MD5SUM=/usr/bin/md5sum
+MKFS=/sbin/mkfs
+MKISOFS=/usr/bin/mkisofs
+MKSWAP=/sbin/mkswap
+SCP=/usr/bin/scp
+SED=/bin/sed
+SSH=/usr/bin/ssh
+SUDO=/usr/bin/sudo
+WGET=/usr/bin/wget
+READLINK=/bin/readlink
+CURL=/usr/bin/curl
+GUNZIP=/bin/gunzip
+BUNZIP2=/bin/bunzip2
 
 # Used for log messages
 SCRIPT_NAME=`basename $0`
@@ -91,17 +94,25 @@ function error_message
 # the command fails
 function exec_and_log
 {
+    # If set to 'true' '$message' will be merged with '$output'
+    mergeoutput=$3
     message=$2
-    output=`$1 2>&1 1>/dev/null`
+    output=`$1 2>&1`
     code=$?
     if [ "x$code" != "x0" ]; then
         log_error "Command \"$1\" failed."
         log_error "$output"
-        if [ -z "$message" ]; then
-            error_message "$output"
-        else
-            error_message "$message"
-        fi
+        case $# in
+            2 )
+               error_message "$message"
+               ;;
+            3 )
+               [ "$mergeoutput" = "true" ] && error_message "$message\n$output"
+               ;;
+            * )
+               error_message "$output"
+               ;;
+        esac
         exit $code
     fi
     log "Executed \"$1\"."
diff --git a/src/tm_mad/ssh/tm_context.sh b/src/tm_mad/ssh/tm_context.sh
index 0827fde..a27d608 100755
--- a/src/tm_mad/ssh/tm_context.sh
+++ b/src/tm_mad/ssh/tm_context.sh
@@ -66,6 +66,7 @@ done
 
 exec_and_log "$MKISOFS -o $TMP_DIR/$DST_FILE -J -R $ISO_DIR" \
     "Error creating iso fs"
+exec_and_log "chmod o-rwx $TMP_DIR/$DST_FILE"
 exec_and_log "$SCP $TMP_DIR/$DST_FILE $DST" \
     "Error copying $TMP_DIR/$DST_FILE to $DST"
 exec_and_log "rm -rf $TMP_DIR" \
diff --git a/src/tm_mad/ssh/tm_mkswap.sh b/src/tm_mad/ssh/tm_mkswap.sh
index b2c9b80..237b59e 100755
--- a/src/tm_mad/ssh/tm_mkswap.sh
+++ b/src/tm_mad/ssh/tm_mkswap.sh
@@ -41,5 +41,5 @@ log "Initializing swap space"
 exec_and_log "$SSH $DST_HOST $MKSWAP $DST_PATH" \
     "Could not create swap on $DST_PATH"
 
-exec_and_log "$SSH $DST_HOST chmod a+w $DST_PATH"
+exec_and_log "$SSH $DST_HOST chmod ug+w,o-rwx $DST_PATH"
 
